arrow-array/src/builder/fixed_size_list_builder.rs

```diff
@@ -194,13 +227,31 @@ where
         );
 
         let nulls = self.null_buffer_builder.finish_cloned();
-        let array_data = ArrayData::builder(DataType::FixedSizeList(
-            Arc::new(Field::new("item", values_data.data_type().clone(), true)),
-            self.list_len,
-        ))
-        .len(len)
-        .add_child_data(values_data)
-        .nulls(nulls);
+
+        let field = match &self.field {
+            Some(f) => {
+                assert_eq!(
+                    f.data_type(),
+                    values_data.data_type(),
+                    "DataType of field ({}) should be the same as the values_builder DataType ({})",
+                    f.data_type(),
+                    values_data.data_type()
+                );
+                if !f.is_nullable() {
+                    assert!(
+                        values_data.null_count() == 0,
```

Comment:
I think it is valid to have a zero length array (which would also have no nulls, but the null count would be zero)


```suggestion
                        values_data.null_count() == 0 || values_data.len() == 0,
```

(same for the check below)